lib/Test2/API.pm

```diff
@@ -596,9 +596,9 @@ generated by the test system:
         my_ok(0, "fail");
     };
 
-    my_ok(@$events == 2, "got 2 events, the pass and the fail");
```

Comment:
That makes sense.  I think you should add a note explaining explicitly, because it seems like a mistake.